### PR TITLE
Integration tests for OTel Collector Attributes.

### PR DESCRIPTION
**Pull request by dengliming**
for [https://github.com/open-telemetry/opentelemetry-go/issues/1035](https://github.com/open-telemetry/opentelemetry-go/issues/1035)

### CODE REVIEW

**Comment by Aneurysm9** [comment]
Please add an entry in `CHANGELOG.md` noting the addition of these tests.

File: CHANGELOG.md [hunk 1]
```diff
@@ -20,6 +20,7 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 - The environment variable used for resource detection has been changed from `OTEL_RESOURCE_LABELS` to `OTEL_RESOURCE_ATTRIBUTES` (#1042)
 - Replace `WithSyncer` with `WithBatcher` in examples. (#1044)
 - Replace the `google.golang.org/grpc/codes` dependency in the API with an equivalent `go.opentelemetry.io/otel/codes` package. (#1046)
+- Integration tests for OTel Collector Attributes. (#1062)
```
Comment (Aneurysm9) [inline CHANGELOG.md:23]:
```suggestion
- Integration tests for more OTel Collector Attribute types. (#1062)
```

I'd also suggest moving this to an `Added` section between the `[Unreleased]` section marker and the `Changed` section marker as these are wholly new integration tests.

Comment (dengliming) [reply CHANGELOG.md]: Thanks. fixed at lastest commit.

File: exporters/otlp/otlp_integration_test.go [hunk 2]
```diff
@@ -426,3 +428,85 @@ func TestNewExporter_withHeaders(t *testing.T) {
...
+	)
+
+	defer func() {
+		_ = exp.Stop()
+	}()
+
+	tp, err := sdktrace.NewProvider(
+		sdktrace.WithConfig(sdktrace.Config{DefaultSampler: sdktrace.AlwaysSample()}),
+		sdktrace.WithBatcher(exp, // add following two options to ensure flush
+			sdktrace.WithBatchTimeout(15),
```
Comment (MrAlias) [inline exporters/otlp/otlp_integration_test.go:454]: Can we attach a unit here?

```suggestion
			sdktrace.WithBatchTimeout(15 * time.Second),
```

File: exporters/otlp/otlp_integration_test.go [hunk 3]
```diff
@@ -426,3 +428,85 @@ func TestNewExporter_withHeaders(t *testing.T) {
...
+	// verification checks of expected data back.
+	_ = mc.stop()
+
+	// Now verify that we only got one span
+	rss := mc.getSpans()
+	if got, want := len(rss), 1; got != want {
+		t.Fatalf("resource span count: got %d, want %d\n", got, want)
+	}
+
+	expecteds := transform.Attributes(testKvs)
```
Comment (MrAlias) [inline exporters/otlp/otlp_integration_test.go:499]: What are we testing here if we are using the same function to verify the result as the one the produced the result? Should this not be replaced with a static slice of expected values?

File: exporters/otlp/otlp_integration_test.go (hunk 2 above)
Comment (dengliming) [reply exporters/otlp/otlp_integration_test.go:454]: Hi~ we only test one span here. `15s ` maybe too long?

File: exporters/otlp/otlp_integration_test.go (hunk 3 above)
Comment (dengliming) [reply exporters/otlp/otlp_integration_test.go]: Thanks. you were right. agree `replace with a static slice of expected values`

File: exporters/otlp/otlp_integration_test.go (hunk 2 above)
Comment (MrAlias) [reply exporters/otlp/otlp_integration_test.go:452]: Gotcha, that was more a suggestion of a unit. Please choose an appropriate one.

Comment (dengliming) [reply exporters/otlp/otlp_integration_test.go]: done. thanks.

**Comment by MrAlias** [comment]
@dengliming I think this looks ready to merged. I tried syncing with upstream, but it looks like upstream changes need to be propagated into these changes. Specifically [the migration of the `api/kv` package into the new `label` package](https://github.com/open-telemetry/opentelemetry-go/pull/1060). LMK if you need any help implementing the changes.

**Comment by dengliming** [comment]
@MrAlias Fixed.

**Pull request merged**